OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1

```diff
@@ -237,7 +262,18 @@ function Import-ServerConfig {
 
     if (Test-OctopusVersionSupportsShowConfiguration) {
         $rawConfig = & $octopusServerExePath show-configuration --format=json-hierarchical --noconsolelogging --console --instance $InstanceName
-        $config = $rawConfig | ConvertFrom-Json
+
+        # handle a specific error where an exception in registry migration finds its way into the json-hierarchical output
+        if(Test-ValidJson $rawConfig) {
+            $config = $rawConfig | ConvertFrom-Json
+        } else {
+            $cleanedUpConfig = Get-CleanedJson $rawConfig
```

Comment:
Maybe stick a `write-warning "show-configuration returned invalid json. Attempting to clean it up."

Reply:
sorted